block-producer/src/state_view/mod.rs

```diff
@@ -54,7 +56,22 @@ where
         &self,
         candidate_tx: &ProvenTransaction,
     ) -> Result<(), VerifyTxError> {
-        // 1. soft-check if `tx` violates in-flight requirements.
+        #[cfg(not(test))]
+        // The verification of the transaction proof is disabled in tests for now because
+        // we are using a `DummyProver`, hence proofs generated in tests are invalid.
+        // TODO: Add valid proof generation in tests
+        {
+            // 1. Verify transaction proof
+            //
+            // This check makes sure that the transaction proof that has been attached to the transaction
+            // is valid, errors out if the proof is invalid.
+            let tx_verifier = TransactionVerifier::new(96);
+            tx_verifier
+                .verify(candidate_tx.clone())
+                .map_err(|_| VerifyTxError::InvalidTransactionProof(candidate_tx.id()))?;
+        }
+
+        // 2. soft-check if `tx` violates in-flight requirements.
```

Comment:
Suggestion: Remove the counter. We don't need to know the step count (doesn't help with understanding things). And having them is just extra work (this clutters the git history, and can easily get out of sync if someone doesn't notice they should be changed).